helix-view/src/editor.rs

```diff
@@ -229,6 +229,113 @@ where
     Ok(chars)
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct CompletionItemKindWrapper(pub CompletionItemKind);
+
+impl From<lsp::CompletionItemKind> for CompletionItemKindWrapper {
+    fn from(value: lsp::CompletionItemKind) -> Self {
+        Self(value)
+    }
+}
+
+impl ToString for CompletionItemKindWrapper {
+    fn to_string(&self) -> String {
+        match self.0 {
+            CompletionItemKind::TEXT => "text",
+            CompletionItemKind::METHOD => "method",
+            CompletionItemKind::FUNCTION => "function",
+            CompletionItemKind::CONSTRUCTOR => "constructor",
+            CompletionItemKind::FIELD => "field",
+            CompletionItemKind::VARIABLE => "variable",
+            CompletionItemKind::CLASS => "class",
+            CompletionItemKind::INTERFACE => "interface",
+            CompletionItemKind::MODULE => "module",
+            CompletionItemKind::PROPERTY => "property",
+            CompletionItemKind::UNIT => "unit",
+            CompletionItemKind::VALUE => "value",
+            CompletionItemKind::ENUM => "enum",
+            CompletionItemKind::KEYWORD => "keyword",
+            CompletionItemKind::SNIPPET => "snippet",
+            CompletionItemKind::COLOR => "color",
+            CompletionItemKind::FILE => "file",
+            CompletionItemKind::REFERENCE => "reference",
+            CompletionItemKind::FOLDER => "folder",
+            CompletionItemKind::ENUM_MEMBER => "enum_member",
+            CompletionItemKind::CONSTANT => "constant",
+            CompletionItemKind::STRUCT => "struct",
+            CompletionItemKind::EVENT => "event",
+            CompletionItemKind::OPERATOR => "operator",
+            CompletionItemKind::TYPE_PARAMETER => "type_parameter",
+            _ => "invalid", // invalid, but should never get one.
+        }
+        .into()
+    }
+}
+
+impl Serialize for CompletionItemKindWrapper {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let name = self.to_string();
+        serializer.serialize_str(&name)
```

Comment:
let's inline `to_string` and use `&str` instead to avoid a pointless allocation

Reply:
Can't inline since this is used in `helix-term/src/ui/completion.rs`, but changing this from `impl ToString` to a `fn name() -> &str` is better